### ISSUE TITLE
[Bug]: Invoke logservice.Client.Read with parameter `firstLsn = 0` would get an error

### ISSUE THREAD

**Issue by cnutshell** [matrixorigin/matrixone#4334]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 6e754c8
- Hardware parameters: CI environment
- OS type: Linux
- Others:
```

### Actual Behavior

Invoke logservice.Client.Read with parameter `firstLsn = 0` would get an error:  `invalid log range`

**Comment by cnutshell** [comment]
<img width="1041" alt="image" src="https://user-images.githubusercontent.com/20291742/182368415-33152245-b852-4a32-b15f-67dde34f3b25.png">

**Comment by lni** [comment]
This seems to be a bug in kv_txn_storage.go - log service uses 1-based indexing, passing a 0 value as the firstIndex parameter will cause the log service to return an invalid log range error. 

Reassigned to @zhangxu19830126

**Comment by cnutshell** [comment]
Issue resolved after PR #4463 merged.
